echopype/calibrate/ek80_complex.py

```diff
@@ -278,11 +323,18 @@ def compress_pulse(backscatter: xr.DataArray, chirp: Dict) -> xr.DataArray:
         # Select channel `chan` and drop the specific beam dimension if all of the values are nan.
         backscatter_chan = backscatter.sel(channel=chan).dropna(dim="beam", how="all")
 
+        # Extract transmit values
         tx = chirp[str(chan.values)]
+
+        # Compute complex conjugate of transmit values and reverse order of elements
         replica = np.flipud(np.conj(tx))
 
+        # Create partial of _convolve_nans_efficiently with passed in replica
+        _convolve_nans_efficiently_partial = partial(_convolve_nans_efficiently, replica=replica)
+
+        # Apply convolve on backscatter (along range sample dimension) and replica
         pc = xr.apply_ufunc(
-            lambda m: (signal.convolve(m, replica, mode="full")[replica.size - 1 :]),
+            _convolve_nans_efficiently_partial,
```

Comment:
I am a little confused by the new function here. Wouldn't it work as well with the following sequence:
- set all NaN elements in `backscatter_chan` to 0
- run the original `xr.apply_ufunc` block
- set all 0 elements in `pc` to NaN

Reply:
Oh yup, you're right, I'll do that instead 👍